### PR TITLE
Add new UIType MultiListFields

### PR DESCRIPTION
**Pull request by adudekyeti**

### CODE REVIEW

File: app/Conditions/QueryFields/MultiListFieldsField.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * MultiListFields Query Field Class.
```
Comment (mariuszkrzaczkowski) [inline app/Conditions/QueryFields/MultiListFieldsField.php]: ?

File: app/Conditions/QueryFields/MultiListFieldsField.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ *
```
Comment (mariuszkrzaczkowski) [inline app/Conditions/QueryFields/MultiListFieldsField.php]: @package

File: app/Conditions/QueryFields/MultiListFieldsField.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+	/**
+	 * Function to get combinations of string from Array.
+	 *
+	 * @param array  $array
+	 * @param string $tempString
+	 *
+	 * @return array
+	 */
+	public function getCombinations($array, $tempString = '')
```
Comment (mariuszkrzaczkowski) [inline app/Conditions/QueryFields/MultiListFieldsField.php]: ?

File: layouts/basic/modules/Vtiger/Edit/Field/MultiListFields.tpl [hunk 4]
```diff
@@ -1,13 +1,13 @@
 {*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
 {strip}
 	<!-- tpl-Base-Edit-Field-MultiListFields -->
-		{assign var=FIELD_INFO value=\App\Purifier::encodeHtml(\App\Json::encode($FIELD_MODEL->getFieldInfo()))}
-		{assign var=PICKLIST_VALUES value=$FIELD_MODEL->getPicklistValues()}
-		{assign var=SPECIAL_VALIDATOR value=$FIELD_MODEL->getValidator()}
-		{assign var=FIELD_VALUE value=$FIELD_MODEL->getEditViewDisplayValue($FIELD_MODEL->get('fieldvalue'),$RECORD)}
+		{assign var="FIELD_INFO" value=\App\Purifier::encodeHtml(\App\Json::encode($FIELD_MODEL->getFieldInfo()))}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Edit/Field/MultiListFields.tpl]: ?

File: layouts/basic/modules/Vtiger/List/Field/MultiListFields.tpl [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+{strip}
+	<!-- tpl-Base-List-Field-MultiListFields -->
+	{assign var="FIELD_INFO" value=\App\Json::encode($FIELD_MODEL->getFieldInfo())}
+	{assign var="PICKLIST_VALUES" value=$FIELD_MODEL->getPicklistValues(true)}
+	{if isset($SEARCH_INFO['searchValue'])}
+		{assign var="SEARCH_VALUES" value=explode('##', \App\Purifier::decodeHtml($SEARCH_INFO['searchValue']))}
+	{else}
+		{assign var="SEARCH_VALUES" value=[]}
+	{/if}
+	<div class="tpl-List-Field-MultiList picklistSearchField">
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/List/Field/MultiListFields.tpl]: ?

File: modules/Vtiger/uitypes/MultiListFields.php [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+			$value = explode(',', $value);
+		}
+		if (!\is_array($value)) {
+			throw new \App\Exceptions\Security('ERR_ILLEGAL_FIELD_VALUE||' . $this->getFieldModel()->getFieldName() . '||' . $this->getFieldModel()->getModuleName() . '||' . $value, 406);
+		}
+		foreach ($value as $item) {
+			if (!\is_string($item)) {
+				throw new \App\Exceptions\Security('ERR_ILLEGAL_FIELD_VALUE||' . $this->getFieldModel()->getFieldName() . '||' . $this->getFieldModel()->getModuleName() . '||' . $value, 406);
+			}
+			if ($item != strip_tags($item)) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/MultiListFields.php]: ?

File: modules/Vtiger/uitypes/MultiListFields.php [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getDisplayValue($value, $record = false, $recordModel = false, $rawText = false, $length = false)
+	{
+		if (empty($value)) {
+			return null;
+		}
+		$value = str_ireplace(',', ', ', trim($value, ','));
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/MultiListFields.php]: :?

File: app/Conditions/QueryFields/MultiListFieldsField.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
```
Comment (rskrzypczak) [inline app/Conditions/QueryFields/MultiListFieldsField.php]: remove function.
Inherit from  MultipicklistField

File: app/Conditions/QueryFields/MultiListFieldsField.php [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+		}
+		return $result ?? [];
+	}
+
+	/**
+	 * Get value.
+	 *
+	 * @return array
+	 */
+	public function getValue(): array
```
Comment (rskrzypczak) [inline app/Conditions/QueryFields/MultiListFieldsField.php]: remove function.
Inherit from MultipicklistField

File: layouts/basic/modules/Vtiger/ConditionBuilder/MultiListFields.tpl [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+{strip}
+	<div class="tpl-Base-ConditionBuilder-MultiListFields">
+		{assign var="PICKLIST_VALUES" value=$FIELD_MODEL->getPicklistValues()}
+		{assign var="FIELD_VALUES" value=explode(',', $VALUE)}
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/ConditionBuilder/MultiListFields.tpl]:
```suggestion
		{assign var="FIELD_VALUES" value=explode('##', $VALUE)}
```

File: modules/Vtiger/uitypes/MultiListFields.php [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+	public function getDbConditionBuilderValue($value, string $operator)
+	{
+		$values = [];
+		if (!\is_array($value)) {
+			$value = $value ? explode('##', $value) : [];
+		}
+		foreach ($value as $val) {
+			$values[] = parent::getDbConditionBuilderValue($val, $operator);
+		}
+		return implode(',', $values);
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/MultiListFields.php]:
```suggestion
		return implode('##', $values);
```

File: modules/Vtiger/uitypes/MultiListFields.php [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+	 */
+	public function getTemplateName()
+	{
+		return 'Edit/Field/MultiListFields.tpl';
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getListSearchTemplateName()
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/uitypes/MultiListFields.php]: ?

**Pull request merged**